crates/std_detect/src/detect/cache.rs

```diff
@@ -81,83 +77,44 @@ impl Initializer {
 }
 
 /// This global variable is a cache of the features supported by the CPU.
+#[cfg(target_pointer_width = "64")]
 static CACHE: Cache = Cache::uninitialized();
 
-/// Feature cache with capacity for `CACHE_CAPACITY` features.
+/// This global variable is a cache of the features supported by the CPU.
+#[cfg(target_pointer_width = "32")]
+static CACHE: [Cache; 2] = [Cache::uninitialized(), Cache::uninitialized()];
```

Comment:
Instead of defining this differently on each platform, could this perhaps be unified a bit like so?

```
#[cfg(target_pointer_width = "64")]
 static CACHE: [Cache; 1] = [Cache::uninitialized()];
#[cfg(target_pointer_width = "32")]
 static CACHE: [Cache; 2] = [Cache::uninitialized(), Cache::uninitialized()];
```

I think that'd help cut down on the `#[cfg]` traffic

Reply:
Awesome hint, thanks!